### DISCUSSION TITLE
Can I control the vertical position of a margin note?

### DISCUSSION THREAD

**Discussion by emmansh** [quarto-dev/quarto-cli#3309] [Q&A]
I want to add a note in the margin, to explain something in a code chunk. However, if the code chunk is lengthy (i.e., > 10 lines), I don't know how to position the margin note to be at a specific height.

I've seen the answer by @dragonstyle [here](https://github.com/quarto-dev/quarto-cli/discussions/2939#discussioncomment-3924936), saying that:
> Right now there isn't a way to customize how we place items in the margin

Still, I'm not sure this statement is relevant to my problem. I'll demonstrate with an example:

<pre>
# .qmd
```{r}
# 1
# 2
# 3
# 4
# 5
# 6
# 7
# 8
# 9
# 10
# 11
# 12
# 13
# 14
# 15
```
::: column-margin
I want to write something in the margin that refers to code in line 8,
and therefore I wish to vertically position this margin note at the same height
as line 8 in the code chunk.
:::
</pre>

And I get this
<img width="590" alt="margin" src="https://user-images.githubusercontent.com/18688877/201657915-d99ede9a-92af-4529-a6f8-7d76fb2e4520.png">

It's not what I want. It's easy to see that the margin note aligns with line `2` in the code chunk. But I want to push it down to align with line 8. Something like that:
<img width="590" alt="margin" src="https://user-images.githubusercontent.com/18688877/201659389-7a4c443b-42ea-4717-97b8-d51e1a491aee.png">

Is there a way to achieve such custom vertical positioning using quarto's features? That is, I *do not* want to hack the HTML/CSS by myself.

**Comment by dragonstyle** [comment]
Without marking up the div that you're placing in the column and adding padding to it using custom css, I can't think of a way to accomplish this, though it would be a very cool feature for native Quarto to support...

**Reply by jcpetitto** [reply]
I wanted to reply just to say I was trying to determine how to do just this, and I came across this post outlining (with an EXCELLENT example) my exact question. FWIW, I would also like to be able to place items in the margin relative to a user-defined target, if you will, in the main text.
